nsxt/resource_nsxt_policy_edge_transport_node.go

```diff
@@ -0,0 +1,1669 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: MPL-2.0
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var policyEdgeNodeFormFactorValues = []string{
+	model.PolicyEdgeTransportNode_FORM_FACTOR_SMALL,
+	model.PolicyEdgeTransportNode_FORM_FACTOR_MEDIUM,
+	model.PolicyEdgeTransportNode_FORM_FACTOR_LARGE,
+	model.PolicyEdgeTransportNode_FORM_FACTOR_XLARGE,
+}
+
+var policySyslogLogLevelValues = []string{
+	model.SyslogConfiguration_LOG_LEVEL_EMERGENCY,
+	model.SyslogConfiguration_LOG_LEVEL_ALERT,
+	model.SyslogConfiguration_LOG_LEVEL_CRITICAL,
+	model.SyslogConfiguration_LOG_LEVEL_ERROR,
+	model.SyslogConfiguration_LOG_LEVEL_WARNING,
+	model.SyslogConfiguration_LOG_LEVEL_NOTICE,
+	model.SyslogConfiguration_LOG_LEVEL_INFO,
+	model.SyslogConfiguration_LOG_LEVEL_DEBUG,
+}
+
+var policySyslogProtocolValues = []string{
+	model.SyslogConfiguration_PROTOCOL_TCP,
+	model.SyslogConfiguration_PROTOCOL_UDP,
+	model.SyslogConfiguration_PROTOCOL_TLS,
+	model.SyslogConfiguration_PROTOCOL_LI,
+	model.SyslogConfiguration_PROTOCOL_LI_TLS,
+}
+
+var policyCpuReservationValues = []string{
+	model.CPUReservation_RESERVATION_IN_SHARES_EXTRA_HIGH_PRIORITY,
+	model.CPUReservation_RESERVATION_IN_SHARES_HIGH_PRIORITY,
+	model.CPUReservation_RESERVATION_IN_SHARES_NORMAL_PRIORITY,
+	model.CPUReservation_RESERVATION_IN_SHARES_LOW_PRIORITY,
+}
+
+var managementAssignments = []string{
+	"dhcp_v4",
+	"static_ipv4",
+	"static_ipv6",
+}
+
+var tepAssignments = []string{
+	"dhcp_v4",
+	"dhcp_v6",
+	"static_ipv4_list",
+	"static_ipv4_pool",
+	"static_ipv6_list",
+	"static_ipv6_pool",
+}
+
+func getPolicyIPAssignmentSchema(required bool, minItems, maxItems int, validAssignments []string) *schema.Schema {
```

Comment:
looks like some assignments - `no_assignment`, mac lists are never used?

Reply:
Also auto_conf - as I already implemented these and since this code could be reused further on, I left these in as the SDK struct supports them. I hid them in the schema and documentation though.

Reply:
Same for mac pairs etc...

Reply:
I'd suggest to add a comment in code about which types are used and which ones are for future use or whatever :)